### ISSUE TITLE
about the disign for LNode

### ISSUE THREAD

**Issue by shiyan224** [Kei18/lacam2#13]
Why does every LNode have to inherit the whole “who” and “where” vectors from its parent?
Instead of that, I think it might be better to just store two integers for “who” and “where” in a single LNode and we could jump to its parent recursively to collect constraints, which is more friendly to memory.

**Comment by Kei18** [comment]
Yes it's doable. But I did not take that way to discard low-level search nodes during the search. Not sure which is effective
